### PR TITLE
Feat/2614 link to cohort builder

### PR DESCRIPTION
**Pull request by adipaul1981**
image
image

### CODE REVIEW

File: src/components/CohortBuilder/SQONProvider.js [hunk 1]
```diff
@@ -112,17 +119,27 @@ class SQONProvider extends React.Component {
...
     });
   }
 }
 
-const mapStateToProps = state => {
+const mapStateToProps = (state) => {
   const { sqons, activeIndex, uid, virtualStudyId } = state.currentVirtualStudy;
+  const { sets } = state.saveSets.userSets;
+
+  let sqonDisctionary = [];
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/SQONProvider.js:131]: On a personal level, I prefer a more functional style. For example: const sqonDictionnary = (sets || []).map(s => ({ setId: s.setId, tag: s.tag}))

Comment (evans-g-crsj) [inline src/components/CohortBuilder/SQONProvider.js:131]: typo in dictionnary

File: src/store/actionCreators/saveSets.ts [hunk 2]
```diff
@@ -230,6 +231,16 @@ export const deleteUserSets = (
   }
 };
 
+export const createQueryInCohortBuilder = (
+  setInfo: SetInfo,
+): ThunkAction<void, RootState, null, SetsActionTypes> => async (dispatch) => {
+  try {
+    dispatch(requestCreateQueryInCohort(setInfo));
+  } catch (e) {
+    console.error(e.message);
```
Comment (evans-g-crsj) [inline src/store/actionCreators/saveSets.ts:240]: what happen when it fails?

File: src/store/reducers/Test/currentVirtualStudy.test.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import reducer from '../currentVirtualStudy';
+import { Action } from 'redux';
+import { requestCreateQueryInCohort } from '../../actionCreators/saveSets';
+import { SetInfo } from '../../../components/UserDashboard/ParticipantSets';
```
Comment (evans-g-crsj) [inline src/store/reducers/Test/currentVirtualStudy.test.tsx:4]: ../../..

File: src/store/reducers/currentVirtualStudy.js [hunk 4]
```diff
@@ -47,6 +48,53 @@ const resetState = (diff = {}) => {
...
+  const sqonExists = sqons.some((s) =>
+    s.content.some((t) => t.content.value === `set_id:${setInfo.key}`),
+  );
+
+  if (sqonExists) {
+    return {
+      newSqons: sqons,
+      activeIndex: sqons.length - 1,
+    };
+  } else {
```
Comment (evans-g-crsj) [inline src/store/reducers/currentVirtualStudy.js:89]: unnecessary else, I think

**Review by evans-g-crsj — APPROVED** [review]

File: src/store/actionCreators/saveSets.ts (hunk 2 above)
Comment (adipaul1981) [reply src/store/actionCreators/saveSets.ts:240]: thats just wrong on my part. it should not be inside a try catch

File: src/store/reducers/currentVirtualStudy.js (hunk 4 above)
Comment (adipaul1981) [reply src/store/reducers/currentVirtualStudy.js:89]: well if the set (sqon) is already in the CB we just return the unaltered sqons. If not we add it. i think it is required.

**Pull request merged**